test/suites/integration/httproute_method_match_test.go

```diff
@@ -89,40 +99,38 @@ var _ = Describe("HTTPRoute method matches", func() {
 
 		//get the pods of deployment1
 		pods := testFramework.GetPodsByDeploymentName(deployment1.Name, deployment1.Namespace)
-		cmd1 := fmt.Sprintf("curl -X GET %s", dnsName)
-		stdout, _, err := testFramework.PodExec(pods[0].Namespace, pods[0].Name, cmd1, true)
-		Expect(err).To(BeNil())
-		Expect(stdout).To(ContainSubstring("test-get handler pod"))
-
-		cmd2 := fmt.Sprintf("curl -X POST %s", dnsName)
-		stdout, _, err = testFramework.PodExec(pods[0].Namespace, pods[0].Name, cmd2, true)
-		Expect(err).To(BeNil())
-		Expect(stdout).To(ContainSubstring("test-post handler pod"))
-
-		invalidCmd := fmt.Sprintf("curl -X DELETE %s", dnsName)
-		stdout, _, err = testFramework.PodExec(pods[0].Namespace, pods[0].Name, invalidCmd, true)
-		Expect(err).To(BeNil())
-		Expect(stdout).To(ContainSubstring("Not Found"))
-
-		testFramework.ExpectDeleted(ctx,
-			gateway,
+
+		Eventually(func(g Gomega) {
+			cmd := fmt.Sprintf("curl -X GET %s", dnsName)
+			stdout, _, err := testFramework.PodExec(pods[0].Namespace, pods[0].Name, cmd, true)
+			g.Expect(err).To(BeNil())
+			g.Expect(stdout).To(ContainSubstring("test-get handler pod"))
+		}).WithTimeout(30 * time.Second).WithOffset(1).Should(Succeed())
+
+		Eventually(func(g Gomega) {
+			cmd := fmt.Sprintf("curl -X POST %s", dnsName)
+			stdout, _, err := testFramework.PodExec(pods[0].Namespace, pods[0].Name, cmd, true)
+			g.Expect(err).To(BeNil())
+			g.Expect(stdout).To(ContainSubstring("test-post handler pod"))
+		}).WithTimeout(30 * time.Second).WithOffset(1).Should(Succeed())
+
+		Eventually(func(g Gomega) {
+			invalidCmd := fmt.Sprintf("curl -X DELETE %s", dnsName)
+			stdout, _, err := testFramework.PodExec(pods[0].Namespace, pods[0].Name, invalidCmd, true)
+			g.Expect(err).To(BeNil())
+			g.Expect(stdout).To(ContainSubstring("Not Found"))
+		}).WithTimeout(30 * time.Second).WithOffset(1).Should(Succeed())
+	})
```

Comment:
I guess we can bear first few `curl` failed due to immature dataplane xds config and eventually expect traffic can work?

Reply:
Yes. There will be a short delay as configuration converges and Lattice is able to process the request successfully. We don't have a good way of programmatically knowing this convergence has happened, so we rely on getting there "Eventually" :)